### PR TITLE
4591 group by item hides batch even if only one line present

### LINKED ISSUE #4591: Group by item hides batch even if only one line present
## Describe the bug
A. 
image
image

We show most other columns with [multiple] if there are are several lines, or if there is just one line we typically show the value for the one line - except for batch, which appears to show nothing?

B.
Pack sell-prices when grouped by item not matching correctly. see below

### Environment

Local on `develop`, though testers have reported on their systems too.

## To Reproduce

Steps to reproduce the behaviour:
### A. Group by item hides batch even if only one line present
1. Go to open an inbound shipment 
2. Add an invoice line that has a batch
3. Toggle group by item on and off, notice that batch is hidden when **on**

### B. `Pack sell price` value is not updated correctly when grouped by item
1. In an Inbound, add multiple lines of same item with different pack-sell prices, sth like this:
image

2. Note the pack-sell prices there
3. Group by item
4. See error: the total sums up to 0; should either be total of all OR just [multiple]
image
 
## Expected behaviour

A. Should show batch if there is just one line and the UI is grouping by item.
B. should either be total of all OR just [multiple]

## Agreed Solution

TBD

## Additional context

Thinking that if the columns aren't that useful showing [multiple] in `group by item`, should we remove the columns? Could show some aggregate columns instead? We already have unit quantity, perhaps swap the costs columns for total cost, total sell price? If wanting broad nitty gritty then don't group by item.

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #4591 and #4918

# 👩🏻‍💻 What does this PR do?
### Inbound Shipment
- Fixes batch not showing up even with only 1 line
- Fixes pack size not showing up with [multiple] for multiple lines 
- Changed sell price per pack to `cost price per unit` to match Outbound Shipment. Makes more sense for Inbound to show cost price than sell price.
- Refactor has2Dp checks + add comments (: 
Screenshot 2024-10-01 at 4 18 28 PM

# 🧪 Testing

- [ ] Create inbound
- [ ] Have multiple lines for one line. Check changed lines mentioned above are correct
- [ ] Have one line: Should see all column info

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [ ] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [x] **These areas should be updated or checked**: 
  1. Screenshots + descriptions
  2.

### CODE REVIEW

File: client/packages/invoices/src/InboundShipment/DetailView/ContentArea/columns.ts [hunk 1]
```diff
@@ -233,35 +239,27 @@ export const useInboundShipmentColumns = () => {
         },
       ],
       {
-        label: 'label.sell',
-        key: 'sellPricePerPack',
+        label: 'label.cost-per-unit',
+        key: 'costPricePerUnit',
```
Comment (roxy-dao) [inline client/packages/invoices/src/InboundShipment/DetailView/ContentArea/columns.ts:243]: Changed this to cost per unit to match with OS sell price per unit... The pack sizes are can be all different, so not sure how useful pack per unit price really is?

File: client/packages/common/src/utils/numbers/NumUtils.ts [hunk 2]
```diff
@@ -29,6 +29,13 @@ export const NumUtils = {
     const multiplier = 10 ** dp;
     return Math.round(value * multiplier) / multiplier;
   },
+  /**
+   * Checks if precision is greater than 2. Some numbers have floating precision
+   * errors e.g. 2.05 * 100 = 204.99999999999997 so using 1000 for a more reliable result...
+   */
+  hasMoreThanTwoDp: (value: number): boolean => {
+    return ((value * 1000) / 10) % 1 !== 0;
+  },
```
Comment (fergie-nz) [inline client/packages/common/src/utils/numbers/NumUtils.ts:38]: Nice! Javascript is weird.

I wonder is this new check 100% safe? or just safer? I guess just safer is fine if it is safe enough?

**Review by fergie-nz — APPROVED** [review]
Nice Roxy looks great to me :)

File: client/packages/common/src/utils/numbers/NumUtils.ts (hunk 2 above)
Comment (roxy-dao) [reply client/packages/common/src/utils/numbers/NumUtils.ts:38]: Safer XD

**Pull request merged**